### PR TITLE
Adds etcd-snapshot subcommands

### PR DESCRIPTION
**Pull request by briandowns**
#### Proposed Changes ####

Adds etcd-snapshot subcommands.

### CODE REVIEW

**Review by dweomer — APPROVED** [review]

File: pkg/cli/cmds/etcd_snapshot.go [hunk 1]
```diff
@@ -2,38 +2,51 @@ package cmds
 
 import (
 	"github.com/rancher/k3s/pkg/cli/cmds"
+	"github.com/rancher/k3s/pkg/cli/etcdsnapshot"
 	"github.com/rancher/rke2/pkg/rke2"
 	"github.com/urfave/cli"
 )
 
+var k3sFlags = map[string]*K3SFlagOption{
```
Comment (brandond) [inline pkg/cli/cmds/etcd_snapshot.go:10]: Why the breakout from the `var ()` block below - is this a gofmt thing?

Comment (brandond) [reply pkg/cli/cmds/etcd_snapshot.go:10]: Oh I see, if you want to reference them in subsequent var definitions you have to define them earlier.

Comment (briandowns) [reply pkg/cli/cmds/etcd_snapshot.go:10]: Accounting for the coming PRs with the rest of the commands.

**Review by brandond — APPROVED** [review]

**Review by Oats87 — APPROVED** [review]

File: pkg/cli/cmds/etcd_snapshot.go [hunk 2]
```diff
@@ -2,38 +2,82 @@ package cmds
...
-		"s3-bucket":          copy,
-		"s3-region":          copy,
-		"s3-folder":          copy,
-	})
+	k3sEtcdSnapshotBase = mustCmdFromK3S(cmds.NewEtcdSnapshotCommand(EtcdSnapshotRun, subcommands), k3sFlags)
 )
 
 func NewEtcdSnapshotCommand() cli.Command {
-	cmd := k3sEtcdSnapshotBase
-	cmd.Flags = append(cmd.Flags, commonFlag...)
```
Comment (Oats87) [inline pkg/cli/cmds/etcd_snapshot.go:35]: Did you mean to omit `commonFlag`?

Comment (briandowns) [reply pkg/cli/cmds/etcd_snapshot.go:35]: Yes, this contains flags unnecessary to the subcommands.

Comment (brandond) [reply pkg/cli/cmds/etcd_snapshot.go:35]: Does look like that should not be dropped

**Review by brandond — CHANGES_REQUESTED** [review]

**Review by brandond — APPROVED** [review]

**Pull request merged**